### PR TITLE
PE-1064

### PR DESCRIPTION
**Pull request by talbaneth**
# Description

# Contribution Checklist

- [ ] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Consider if this PR needs the `officeHours` modifier override
- [ ] Verify expiration (`30 days` unless otherwise specified)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in Goerli changelog or known
- [ ] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Ensure contract is verified on `Goerli` etherscan
- [ ] Change test to use Goerli spell address and deploy timestamp
- [ ] Cast spell on Goerli `make spell="0x-deployed-spell-address" cast-spell`
- [ ] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

### CODE REVIEW

File: src/Goerli-DssSpell.t.sol [hunk 1]
```diff
@@ -464,136 +464,4 @@ contract DssSpellTest is GoerliDssSpellTestBase {
 
         assertEq(vest.rxd(1), WAD);
     }
```
Comment (naszam) [inline src/Goerli-DssSpell.t.sol:466]: Consider cleanup `testNewChainlogValues` and `testNewIlkRegistryValues` and set visibility to `private` to disable.

**Review by naszam — COMMENTED** [review]
Checked:

- Office Hours 
    - Off (default on Goerli)
- Exec Hash 
    - empty (default on Goerli)
- 30 Days Expiry
    - weekly
- Pragma OK
    - No Collateral Onboarding
- Rates OK (match IPFS table and script)
- Math OK
    - Units
        - `MILLION = 10**6`
- Risk Parameters Changes Match Doc (`jug.ilk.duty`, `autoLine` DC)
- Spell Actions Look Good
    - PPG - Maker Open Market Committee Option 2 (includes Option 1)
- DssExecLib `v0.0.9`
- Test Coverage Looks Good
    - `config`
- Test PASS (commit ab87895)
```haskell
dapp-test: rpc block: latest
Running 16 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 7139)
[PASS] test_auth_in_sources() (gas: 5422616)
[PASS] test_use_eta() (gas: 343454)
[PASS] testOnTime() (gas: 725349)
[PASS] test_nextCastTime() (gas: 344574)
[PASS] test_chainlog_values() (gas: 5718002)
[PASS] testFailTooEarly() (gas: 4119)
[PASS] test_bytecode_matches() (gas: 1263364)
[PASS] test_chainlog_version_bump() (gas: 3551718)
[PASS] test_auth() (gas: 21615633)
[PASS] testSpellIsCast_GENERAL() (gas: 22198700)
[PASS] testFailWrongDay() (gas: 4141)
[PASS] testCastCost() (gas: 727157)
[PASS] testNewChainlogValues() (gas: 883513)
[PASS] testNewIlkRegistryValues() (gas: 758612)
[PASS] testFailTooLate() (gas: 4140)

Running 1 tests for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 915407)
```

File: src/Goerli-DssSpell.t.sol (hunk 1 above)
Comment (talbaneth) [reply src/Goerli-DssSpell.t.sol:454]: Done in [7900c26](https://github.com/makerdao/spells-goerli/pull/104/commits/7900c2635a91d18e257ccc5e89e98ce865e05c6b)

File: src/Goerli-DssSpell.t.sol [hunk 2]
```diff
@@ -159,43 +159,31 @@ contract DssSpellTest is GoerliDssSpellTestBase {
...
-        checkChainlogKey("OPTIMISM_ESCROW");
-        checkChainlogKey("OPTIMISM_GOV_RELAY");
-        checkChainlogKey("ARBITRUM_TELEPORT_BRIDGE");
-        checkChainlogKey("ARBITRUM_TELEPORT_FEE");
-        checkChainlogKey("ARBITRUM_DAI_BRIDGE");
-        checkChainlogKey("ARBITRUM_ESCROW");
-        checkChainlogKey("ARBITRUM_GOV_RELAY");
-        checkChainlogVersion("1.14.0");
+        // checkChainlogKey("MCD_JOIN_TELEPORT_FW_A");
+        // checkChainlogVersion("1.14.0");
```
Comment (naszam) [inline src/Goerli-DssSpell.t.sol:169]: nit: don't need to comment here as visibility is `private`

File: src/Goerli-DssSpell.t.sol [hunk 3]
```diff
@@ -159,43 +159,31 @@ contract DssSpellTest is GoerliDssSpellTestBase {
...
-        assertEq(reg.symbol("TELEPORT-FW-A"), "");
+        // assertEq(reg.pos("TELEPORT-FW-A"), 52);
+        // assertEq(reg.join("TELEPORT-FW-A"), addr.addr("MCD_JOIN_TELEPORT_FW_A"));
+        // assertEq(reg.gem("TELEPORT-FW-A"), address(0));
+        // assertEq(reg.dec("TELEPORT-FW-A"), 0);
+        // assertEq(reg.class("TELEPORT-FW-A"), 4);
+        // assertEq(reg.pip("TELEPORT-FW-A"), address(0));
+        // assertEq(reg.xlip("TELEPORT-FW-A"), address(0));
+        // assertEq(reg.name("TELEPORT-FW-A"), "");
+        // assertEq(reg.symbol("TELEPORT-FW-A"), "");
```
Comment (naszam) [inline src/Goerli-DssSpell.t.sol:186]: Same as above

**Comment by The-Arbiter** [comment]
Review in progress:

Goerli Spell Actions:
- [Skipped] Sets CRVV1ETHSTETH-A stability fee to 2.0%
- Sets MANA-A stability fee to 4.5%
- Sets ETH-A stability fee to 1.5%
- Sets ETH-B stability fee to 3.0%
- Sets WSTETH-A stability fee to 1.5%
- Sets WBTC-A stability fee to 2.0%
- Sets WBTC-B stability fee to 3.5%
- Sets RENBTC-A stability fee to 2.25%
- Sets WSTETH-B automatic debt ceiling limit to $200m 
- [Skipped] Sets CRVV1ETHSTETH-A automatic debt ceiling parameters to $20m (max DC) $10m (max increase per IAM call) and 8 hours (minimum time between IAM calls)

**Checklist**
✅ Office hours returns false
✅ No exec hash for goerli
✅ Constructor has `block.timestamp + 30 days` as expiry time
✅ Rates match IPFS and CLI
✅ Math - only units used are for AutoLine ceilings and params, passed in as "Millions" is correct and ExecLib already scales by Rad, this looks fine. Numbers all in range.
✅ Line -> Autoline -> vat.ilk.line should be the IAM gap value -> WSTETH-B ilk line is 15m (as Rad) but existing IAM gap is 30m on Goerli. Raised with Naz and marking this as fine. Impacts on state reflect intention.
✅ Jug ilk duty check for stability fee updates -> impacts on state reflect intention (rabbit-holed through the code until state changes, all looks fine)
✅ Autoline Params -> All valid 
✅ Spell actions all match [here](https://forum.makerdao.com/t/parameter-changes-proposal-ppg-omc-001-25-august-2022/17448) except for CRVV1ETHSTETH-A, which was discussed within PECU
📝 Tests (TBC, will confirm once this works)
✅ DssExecLib uses c70218a not c0d3c6 (0.0.9 release) but Naz has explained why this is the case.

**Review by hexonaut — DISMISSED** [review]
Good to deploy / cast. Checked:

 * Rates are good
 * Office hours off
 * Stability fees are good
 * Tests look good
 * Tests passing (except Starknet one)

**Comment by naszam** [comment]
Starknet Core seems have been upgraded recently to the following implementation: [0x70c8a579ad08339cca19d77d8646f4b6f0fd098a](https://goerli.etherscan.io/address/0xde29d060D45901Fb19ED6C6e959EB22d8626708e#readContract).
Tests PASS locally, with the `core_implementation` update in `starknet.t.sol`:
```haskell
dapp-test: rpc block: 7534026
Running 1 tests for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 914601)
```

**Review by naszam — DISMISSED** [review]
Test PASS (commit 7900c26):
```haskell
dapp-test: rpc block: 7534026
Running 14 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 7139)
[PASS] test_auth_in_sources() (gas: 5421786)
[PASS] test_use_eta() (gas: 343454)
[PASS] testOnTime() (gas: 724565)
[PASS] test_nextCastTime() (gas: 344596)
[PASS] test_chainlog_values() (gas: 5717129)
[PASS] testFailTooEarly() (gas: 4141)
[PASS] test_bytecode_matches() (gas: 1263386)
[PASS] test_chainlog_version_bump() (gas: 3550934)
[PASS] test_auth() (gas: 21614825)
[PASS] testSpellIsCast_GENERAL() (gas: 22197916)
[PASS] testFailWrongDay() (gas: 4163)
[PASS] testCastCost() (gas: 726373)
[PASS] testFailTooLate() (gas: 4140)
```

LGTDeploy

**Comment by talbaneth** [comment]
Awaiting approvals to cast and archive.

**Review by naszam — APPROVED** [review]
Checked:

- [Diffchecked](https://www.diffchecker.com/VpOKTbDk) `starknet.t.sol` against `spells-mainnet`, Match
- Deployed Spell Verified
    - Optimization Enabled: No
    - Other Settings: default evmVersion, GNU AGPLv3 license
- Deployed Spell Code Matches GitHub
    - [Diffchecked](https://www.diffchecker.com/OS6xKSIW) agaist spell pr (ethescan code matches pr)
- Deploy Timestamp Match (via script)
- DssExecLib `v0.0.9`
- CI PASS (on #105)
- Test PASS (with deployed spell at commit 62a58df ):
```haskell
dapp-test: rpc block: latest
Running 14 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 12139)
[PASS] test_auth_in_sources() (gas: 5433516)
[PASS] test_use_eta() (gas: 350454)
[PASS] testOnTime() (gas: 734295)
[PASS] test_nextCastTime() (gas: 351596)
[PASS] test_chainlog_values() (gas: 5728859)
[PASS] testFailTooEarly() (gas: 9141)
[PASS] test_bytecode_matches() (gas: 1272886)
[PASS] test_chainlog_version_bump() (gas: 3562664)
[PASS] test_auth() (gas: 21626555)
[PASS] testSpellIsCast_GENERAL() (gas: 23468103)
[PASS] testFailWrongDay() (gas: 9163)
[PASS] testCastCost() (gas: 736103)
[PASS] testFailTooLate() (gas: 9140)

Running 1 tests for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 924073)
```

LGTCast

**Review by hexonaut — APPROVED** [review]
Good to cast / archive pending tests from another team member (can’t check as I’m on a plane for 10 hours). Checked:

 * Spell address and timestamp are good
 * Code matches
 * DssExecLib matches repo

**Comment by The-Arbiter** [comment]
Pass

Tests verified as working on PE-1064 (previous result was from forge-ci).

**Comment by talbaneth** [comment]
@naszam @The-Arbiter @hexonaut awaiting approval to merge.

**Review by hexonaut — APPROVED** [review]
Archive matches src. Good to merge.

**Review by naszam — APPROVED** [review]
Checked:

- Archive Match (via script)

LGTMerge

**Pull request merged**